roles/provision-keycloak-apb/tasks/main.yml

```diff
@@ -1,12 +1,41 @@
 - include_tasks: label-serviceinstance.yml
 
 - include_tasks: provision-postgres.yml
-  when: POSTGRESQL_EPHEMERAL is undefined
+  when: POSTGRESQL_EPHEMERAL is undefined or
+        not POSTGRESQL_EPHEMERAL
 
 - include_tasks: provision-postgres-ephemeral.yml
-  when: POSTGRESQL_EPHEMERAL is defined
+  when: POSTGRESQL_EPHEMERAL is defined or
+        POSTGRESQL_EPHEMERAL
 
 - include_tasks: provision-keycloak.yml
+  when: SHARED_SERVICE is not defined or
+        not SHARED_SERVICE
+
+- name: "Retrieve route to keycloak"
```

Comment:
The host can be provided during the provision, I am not seeing where we take advantage of that